README.rst

```diff
@@ -2,56 +2,129 @@
+
+2) "hard nurse" constraint: requires that no nurse works two or more 
+   consecutive days.
+
+3) "soft nurse" constraint: promotes that all nurses should have roughly
+   even work schedules.
+
+This demo seeks to obtain reasonable results for a nurse schedule, based on
+INH's model. Our implementation attempts to find a schedule for a number ``n_nurses`` of nurses and a number ``n_days`` of days that satisfies the following conditions:
+
+* One, and only one, nurse has been assigned to each day (hard shift 
+  constraint)
+* No nurse works two days in a row (hard nurse constraint)
+* The nurses should work the same number of days
+
+Running the demo results in the following output, at the command-line:
+::
+  Size  33
+
+  Energy  0.5999999999999694
+
+  Checking Hard nurse constraint  0.0
+
+  Checking Hard shift constraint  0.0
+
+  Checking Soft nurse constraint  0.6
+
+========= =  =  =  =  =  =  =  =  =  =  ==
+   Day    0  1  2  3  4  5  6  7  8  9  10
+========= =  =  =  =  =  =  =  =  =  =  ==
+Nurse  0           X     X     X     X   
+Nurse  1     X        X     X           X
+Nurse  2  X     X                 X      
+========= =  =  =  =  =  =  =  =  =  =  ==
+
+The results show the following:
+
+* One, and only one, nurse has been assigned to each day
+* No nurse works two days in a row
+* Two nurses work 4 days, and one works three days. Because two nurses work
+  one extra day each, the soft nurse constraint energy is nonzero. Each nurse 
+  working one extra day contributes a total of gamma to the energy. Since
+  gamma is 0.3, the total energy is expected to be 0.6.
 
 Usage
 -----
 
-A simple command that runs the demo. For example,
+To run the demo, run the command
 
 .. code-block:: bash
 
-  python <demo_name>.py
+  python nurse_scheduling.py
 
 
 Code Overview
 -------------
 
-A general overview of how the code works.
+Here is a general overview of the Nurse Scheduling code:
 
-Prefer bite-sized descriptions in bullet points:
-
-* Here's an example bullet point
+* Assign the size of the problem (number of nurses and days) and parameters
+* Compute the "penalty matrix" J
+* Develop the QUBO matrix
+* Run the problem (solve the QUBO)
+* Calculate the hard nurse constraint sum, to check if the hard nurse
+  constraints are satisfied
+* Calculate the hard shift constraint sum, to check if the hard shift
+  constraints are satisfied
+* Calculate the soft nurse constraint sum, to check if the soft nurse
+  constraints are satisfied
+* Print the nurse schedule
 
+Note that the total of the three constraint sums should equal the energy.
 
 Code Specifics
 --------------
 
-Notable parts of the code implementation.
+Some notes on the code:
 
-This is the place to:
+* We use a vector to store the two-dimensional (nurse, day) matrix:
 
-* Highlight a part of the code implementation
-* Talk about unusual or potentially difficult parts of the code
-* Explain a code decision
+  (0, 0) (0, 1) (0, 2)... (0, D) (1, 0) (1, 1)... (1, D)
```

Comment:
I am interpreting "We use a vector to store the two-dimensional (nurse, day) matrix" as "we are flattening the matrix and storing it into a vector". However, we don't actually have this vector that stores `(0, 0) (0, 1) ..`, instead we have indices that follow the order that you have described.

Would the following be accurate?
```
* We store our QUBO in the form of a matrix. The rows and columns of said
  matrix each correspond to different combinations of nurses and days. The
  rows and columns use the following order, where each tuple indicates the
  combination (<nurse>, <day>):
  (0, 0) (0, 1) .. (0, n_days) (1, 0) .. (1, n_days) .. (n_nurses, n_days)
```

Reply:
I used different wording, but thank you for catching this. Please see if it is better.